Makefile.pre.in

```diff
@@ -715,6 +718,16 @@ check-clean-src:
 		exit 1; \
 	fi
 
+# Check that the app store compliance patch can be applied (if configured).
+# This is checked as a dry-run against the original library sources;
+# the patch will be actually applied during the install phase.
+.PHONY: check-app-store-compliance
+check-app-store-compliance:
+	@if [ "$(APP_STORE_COMPLIANCE_PATCH)" != "" ]; then \
+		patch --dry-run --quiet --force --strip 1 --directory "$(abs_srcdir)" --input "$(abs_srcdir)/$(APP_STORE_COMPLIANCE_PATCH)"; \
```

Comment:
Some explanation of the choices here:
* `--dry-run` means no changes are applied
* `--quiet` means no "patching xyz.py" messages are output
* `--force` means no questions are asked of the user, and the patch is assumed to be forward.